src/Security/SslContext.cs

```diff
@@ -523,7 +523,7 @@ public SslStatus SetEncryptionCertificate (SecIdentity identify, IEnumerable<Sec
 			get {
 				IntPtr value;
 				result = SSLCopyPeerTrust (Handle, out value);
-				return (value == IntPtr.Zero) ? null : new SecTrust (value);
+				return (value == IntPtr.Zero) ? null : new SecTrust (value, true);
```

Comment:
Looking at SecTrust, passing true to the owns parameter prevents SecTrust from retaining the handle, but in Dispose, it calls release as long as handle is non-zero. What prevents that from releasing a handle SecTrust didn't retain?